public/app/features/alerting/unified/components/amroutes/MuteTimingsTable.tsx

```diff
@@ -0,0 +1,149 @@
+
+interface Props {
+  alertManagerSourceName: string;
+  muteTimingNames?: string[];
+  hideActions?: boolean;
+}
+
+export const MuteTimingsTable: FC<Props> = ({ alertManagerSourceName, muteTimingNames, hideActions }) => {
+  const styles = useStyles2(getStyles);
+  const dispatch = useDispatch();
+  const amConfigs = useUnifiedAlertingSelector((state) => state.amConfigs);
+  const [muteTimingName, setMuteTimingName] = useState<string>('');
+  const { result }: AsyncRequestState<AlertManagerCortexConfig> =
+    (alertManagerSourceName && amConfigs[alertManagerSourceName]) || initialAsyncRequestState;
+
+  const items = useMemo((): Array<DynamicTableItemProps<MuteTimeInterval>> => {
+    const muteTimings = result?.alertmanager_config?.mute_time_intervals ?? [];
+    return muteTimings
+      .filter(({ name }) => (muteTimingNames ? muteTimingNames.includes(name) : true))
+      .map((mute) => {
+        return {
+          id: mute.name,
+          data: mute,
+        };
+      });
+  }, [result?.alertmanager_config?.mute_time_intervals, muteTimingNames]);
+
+  const columns = useColumns(alertManagerSourceName, hideActions, setMuteTimingName);
+
+  return (
+    <div>
+      {!hideActions && <h5>Mute timings</h5>}
+      {items.length > 0 ? <DynamicTable items={items} cols={columns} /> : <p>No mute timings configured</p>}
+      {!hideActions && (
+        <ConfirmModal
+          isOpen={!!muteTimingName}
+          title="Delete mute timing"
+          body="Are you sure you would like to delete this mute timing"
+          confirmText="Delete"
+          onConfirm={() => dispatch(deleteMuteTimingAction(alertManagerSourceName, muteTimingName))}
+          onDismiss={() => setMuteTimingName('')}
+        />
+      )}
+      {!hideActions && (
+        <LinkButton
+          className={styles.addMuteButton}
+          variant="secondary"
+          href={makeAMLink('alerting/routes/mute-timing/new', alertManagerSourceName)}
+        >
+          Add mute timing
+        </LinkButton>
+      )}
+    </div>
+  );
+};
+
+function useColumns(alertManagerSourceName: string, hideActions = false, setMuteTimingName: (name: string) => void) {
+  return useMemo((): Array<DynamicTableColumnProps<MuteTimeInterval>> => {
+    const columns: Array<DynamicTableColumnProps<MuteTimeInterval>> = [
+      {
+        id: 'name',
+        label: 'Name',
+        renderCell: function renderName({ data }) {
+          return data.name;
+        },
+        size: '250px',
+      },
+      {
+        id: 'timeRange',
+        label: 'Time range',
+        renderCell: ({ data }) => parseTimings(data.time_intervals),
+      },
+    ];
+    if (!hideActions) {
+      columns.push({
+        id: 'actions',
+        label: 'Actions',
+        renderCell: function renderActions({ data }) {
+          return (
+            <div>
+              <Link href={makeAMLink(`/alerting/routes/mute-timing/${btoa(data.name)}/edit`, alertManagerSourceName)}>
```

Comment:
I'm still not certain about whether this is a good path forward. `btoa` throws an exception for unicode characters. I'm thinking of refactoring this to add it as a query param instead and using `encodeURIComponent`